src/RA_StringUtils.cpp

```diff
@@ -83,18 +83,24 @@ std::string Narrow(const std::string& str)
 _Use_decl_annotations_
 std::string& TrimLineEnding(std::string& str) noexcept
 {
-    auto nIndex = str.length();
-    if (nIndex > 0 && str[nIndex - 1] == '\n')
-        nIndex--;
-    if (nIndex > 0 && str[nIndex - 1] == '\r')
-        nIndex--;
-
-    if (nIndex != str.length())
-        str.resize(nIndex);
-
+    auto sCpy{ str };
+    str = TrimLineEnding(std::move_if_noexcept(sCpy));
     return str;
 }
 
+std::string TrimLineEnding(std::string&& str) noexcept
+{
+    auto sRet{ std::move_if_noexcept(str) };
+    
+    // using the .back() and .pop_back() strategy doesn't work
+    if (const auto pos = sRet.rfind('\n', sRet.length()); pos != std::string::npos)
```

Comment:
This is wrong. We only want to truncate the newline character if it is the _last_ character of the string. `rfind` may return one in the middle of the string. I'm pretty sure the "pos" parameter is where it starts searching backwards from, not where it stops.